data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/main/ConsumerVerticleContext.java

```diff
@@ -0,0 +1,234 @@
+
+    this.egress = egress;
+
+    if (egress.hasEgressConfig()) {
+      this.egressConfig = egress.getEgressConfig();
+    } else {
+      this.egressConfig = resource.getEgressConfig();
+    }
+
+    final var maxProcessingTime = maxProcessingTimeMs(consumerConfigs, egressConfig);
+    consumerConfigs.put(ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG, maxProcessingTime);
+  }
+
+  public ConsumerVerticleContext withAuthProvider(final AuthProvider authProvider) {
+    this.authProvider = authProvider;
+    return this;
+  }
+
+  public ConsumerVerticleContext withMeterRegistry(MeterRegistry metricsRegistry) {
+    this.metricsRegistry = metricsRegistry;
+    return this;
+  }
+
+  public ConsumerVerticleContext withWebClientOptions(final WebClientOptions webClientOptions) {
+    this.webClientOptions = new WebClientOptions(webClientOptions);
+    return this;
+  }
+
+  public ConsumerVerticleContext withConsumerFactory(final ConsumerFactory<Object, CloudEvent> consumerFactory) {
+    this.consumerFactory = consumerFactory;
+    return this;
+  }
+
+  public ConsumerVerticleContext withProducerFactory(final ProducerFactory<String, CloudEvent> producerFactory) {
+    this.producerFactory = producerFactory;
+    return this;
+  }
+
+  public DataPlaneContract.Resource getResource() {
+    return resource;
+  }
+
+  public DataPlaneContract.Egress getEgress() {
+    return egress;
+  }
+
+  public int getMaxPollRecords() {
+    if (this.maxPollRecords == null) {
+      final var mpr = getConsumerConfigs().get(ConsumerConfig.MAX_POLL_RECORDS_CONFIG);
+      if (mpr == null) {
+        this.maxPollRecords = DEFAULT_MAX_POLL_RECORDS;
+      } else {
+        this.maxPollRecords = Integer.parseInt(mpr.toString());
+      }
+    }
+    return this.maxPollRecords;
+  }
+
+  public DataPlaneContract.EgressConfig getEgressConfig() {
+    return egressConfig;
+  }
+
+  public AuthProvider getAuthProvider() {
+    return authProvider;
+  }
+
+  public MeterRegistry getMetricsRegistry() {
+    return metricsRegistry;
+  }
+
+  public Map<String, Object> getConsumerConfigs() {
+    return consumerConfigs;
+  }
+
+  public Map<String, Object> getProducerConfigs() {
+    return producerConfigs;
+  }
+
+  public WebClientOptions getWebClientOptions() {
+    return webClientOptions;
+  }
+
+  private synchronized ConsumerVerticleLoggingContext getLoggingContext() {
+    if (loggingContext == null) {
+      loggingContext = new ConsumerVerticleLoggingContext(this);
+    }
+    return loggingContext;
+  }
+
+  public Object getLoggingKeyValue() {
+    return keyValue("context", getLoggingContext());
+  }
+
+  public ConsumerFactory<Object, CloudEvent> getConsumerFactory() {
+    return this.consumerFactory;
+  }
+
+  public ProducerFactory<String, CloudEvent> getProducerFactory() {
+    return this.producerFactory;
+  }
+
+  private static int maxProcessingTimeMs(final Map<String, Object> consumerConfigs, final DataPlaneContract.EgressConfig egressConfig) {
```

Comment:
this still has quiet some logic :sweat_smile:

Reply:
Definitely we can keep splitting.